src/back_end/services/userService.js

```diff
@@ -0,0 +1,153 @@
+                }
+                if (!user.validPassword(password))
+                    return done(null, false, req.flash('loginMessage', 'Oops! Wrong password.'));
+                return done(null, user);
+            });
+    }));
+    passport.use('local-register', new LocalStrategy({
+            usernameField : 'email',
+            passwordField : 'password',
+            passReqToCallback : true
+        },
+        function(req, email, password, done) {
+            User.findOne({ 'local.email' :  email }, function(err, user) {
+                if (err) {
+                    return done(err);
+                }
+                if (user) {
+                    return done(null, false, req.flash('signupMessage', 'That email is already taken.'));
+                } else {
+                    var newUser = new User();
+                    newUser.local.email    = email;
+                    newUser.local.password = newUser.generateHash(password);
+                    newUser.save(function(err) {
+                        if (err) {
+                            logger.logError(err);
+                        }
+                        return done(null, newUser);
+                    });
+                }
+            });
+    }));
+    passport.use('facebook', new FacebookStrategy({
+            clientID : config.auth.facebook.clientID,
+            clientSecret : config.auth.facebook.clientSecret,
+            callbackURL : config.auth.facebook.callbackURL
+        },
+        function(token, refreshToken, profile, done) {
+            process.nextTick(function() {
+                User.findOne({ 'facebook.id' : profile.id }, function(err, user) {
+                    if (err)
+                        return done(err);
+                    if (user) {
+                        return done(null, user);
+                    } else {
+                        var newUser = new User();
+                        newUser.facebook.id    = profile.id;
+                        newUser.facebook.token = token;
+                        newUser.facebook.name  = profile.name.givenName + ' ' + profile.name.familyName;
+                        newUser.facebook.email = profile.emails[0].value;
+                        newUser.save(function(err) {
+                            if (err)
+                                throw err;
+                            return done(null, newUser);
+                        });
+                    }
+
+                });
+            });
+    }));
+
+    var login = function (req, res, next) {
+        passport.authenticate('local-login', {
+            successRedirect : '/profile',
+            failureRedirect : '/login',
+            failureFlash : true
+        })(req, res, next);
+    };
+    var register = function (req, res, next) {
+        passport.authenticate('local-register', {
+            successRedirect : '/profile',
+            failureRedirect : '/signup',
```

Comment:
Let's move list of routes to contstants, cause they are used not only in `app.js` or etc.